build_tools/cmake/iree_macros.cmake

```diff
@@ -58,27 +58,35 @@ endfunction()
 #   runtime/src/iree/base/CMakeLists.txt -> iree::base
 #   tests/e2e/CMakeLists.txt -> iree::tests::e2e
 function(iree_package_ns PACKAGE_NS)
-  # Get the relative path of the current dir (i.e. runtime/src/iree/vm).
-  string(REPLACE ${IREE_ROOT_DIR} "" _RELATIVE_PATH ${CMAKE_CURRENT_LIST_DIR})
-  string(SUBSTRING ${_RELATIVE_PATH} 1 -1 _RELATIVE_PATH)
-
-  # If changing the directory/package mapping rules, please also implement
-  # the corresponding rule in:
-  #   build_tools/bazel_to_cmake/bazel_to_cmake_targets.py
-  # Some sub-trees form their own roots for package purposes. Rewrite them.
-  if(_RELATIVE_PATH MATCHES "^compiler/src/(.*)")
-    # compiler/src/iree/compiler -> iree/compiler
-    set(_PACKAGE "${CMAKE_MATCH_1}")
-  elseif(_RELATIVE_PATH MATCHES "^runtime/src/(.*)")
-    # runtime/src/iree/base -> iree/base
-    set(_PACKAGE "${CMAKE_MATCH_1}")
-  elseif(_RELATIVE_PATH MATCHES "^tools$")
-    # Special case for tools/ -> "" (empty string)
-    # For example, tools/iree-compile -> iree-compile (no namespace)
-    set(_PACKAGE "")
+  if(${IREE_ROOT_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
+    # If IREE_ROOT_DIR is not set to be where CMake config is called, the macro
+    # is used out-of-tree and the relative path should be used directly.
+    string(REPLACE ${CMAKE_SOURCE_DIR} "" _RELATIVE_PATH ${CMAKE_CURRENT_LIST_DIR})
+    string(SUBSTRING ${_RELATIVE_PATH} 1 -1 _RELATIVE_PATH)
+    set(_PACKAGE "${_RELATIVE_PATH}")
   else()
```

Comment:
Ohhh.. hmm.. so you're actually including the `iree_copts.cmake` file [again] yourself:
```cmake
include(${IREE_SOURCE_DIR}/build_tools/cmake/iree_copts.cmake)
```

that's sketchy... can you instead just rely on
```cmake
add_subdirectory(${IREE_SOURCE_DIR} iree)
```

and then use the functions that way?

I was imagining keeping the rest of the branches as-is, but adding a first check for the relative path, e.g.
```
if path isn't under IREE root:
  set namespace based on purely relative path
else if path matches compiler/src:
  ...
else if path matches runtime/src:
  ...
endif
```

vs how you have this now:
```
if root mismatches (here be dragons):
  set namespace based on purely relative path
else:
  if path matches compiler/src:
    ...
  else if path matches runtime/src:
    ...
  endif
endif
```

Reply:
The first two settings serve two different purposes. The first one includes all the compilation options IREE expect the targets to comply, while the second one include IREE as a submodule/library.

I can change the if/else logic to be more resemble to the one you proposed though. Thanks for the feedback.

Reply:
actually, presubmit of this draft fails (hangs forever), and the logic I put here doesn't really make sense. I will comeback with a better strategy. :(

Reply:
> actually, presubmit of this draft fails (hangs forever)

That's unrelated to your change. The early migration to the `iree-org` repository broke all of our CI pipelines. We're expecting to have them back up tomorrow.

Reply:
I see, but my local run of the CMake build also fails with the logic in this PR

```
if(${IREE_ROOT_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
```
is also true for in-project build, so it will break all the existing build scripts in presumbit. Sorry.